src/app/pages/onboarding/sign-in/sign-in.tsx

```diff
@@ -83,96 +40,151 @@ export function SignIn() {
     void submitMnemonicForm(key);
   }
 
+  const mnemonicFieldArray = createNullArrayOfLength(twentyFourWordMode ? 24 : 12);
+
+  // set initialValues to avoid throwing uncontrolled inputs error
+  const initialValues = mnemonicFieldArray.reduce(
+    (accumulator, _, index) => ((accumulator[`${index + 1}`] = ''), accumulator),
+    {}
+  );
+
   return (
-    <Flex
-      flexDirection={['column', 'column', 'column', 'row']}
-      mt={['space.05', 'space.06']}
-      pb="loose"
-      px={['space.05', 'space.05', 'space.11']}
-      width="100%"
-      gap={['space.03', 'space.09']}
-      height="fit-content"
-    >
-      <Stack flex="1" gap="space.06">
-        <styled.h1 textStyle={['heading.03', 'display.02']}>Sign in with your Secret Key</styled.h1>
-        <Box>
-          <styled.p textStyle={['label.01', 'heading.05']} mb="space.02">
-            Enter your Secret Key to sign in with an existing wallet
-          </styled.p>
-
-          <styled.span textStyle="body.02">
-            Tip: You can paste in your entire Secret Key at once
-          </styled.span>
-        </Box>
-        <LeatherButton
-          mt="space.03"
-          variant="link"
-          textStyle="caption.01"
-          width="fit-content"
-          onClick={() => {
-            setTwentyFourWordMode(!twentyFourWordMode);
-            setMnemonic(createNullArrayOfLength(twentyFourWordMode ? 24 : 12));
-          }}
-        >
-          {twentyFourWordMode ? 'Have a 12-word Secret Key?' : 'Use 24 word Secret Key'}
-        </LeatherButton>
-      </Stack>
-      <Stack
-        borderRadius="16px"
-        backgroundColor="brown.1"
-        px={['', 'space.05']}
-        pt={['space.02', 'space.07']}
-        pb={['space.02', 'space.05']}
-        flex="1"
-        gap="space.04"
+    <>
+      <Flex
+        flexDirection={['column', 'column', 'column', 'row']}
+        mt={['space.05', 'space.06']}
+        pb="loose"
+        px={['space.05', 'space.05', 'space.11']}
```

Comment:
Would love to see this page use the `.layout.tsx` pattern, so this component can focus on the logic and not the layout.

Reply:
I think this is OK now with the layout component?